### DISCUSSION TITLE
Compilation of any scrartcl document to PDF results in the error: 'Argument of \strip@prefix has an extra } '

### DISCUSSION THREAD

**Discussion by Niklewa** [quarto-dev/quarto-cli#8827] [Q&A] [answered]
### Description

I have the following problem: any document that is set to render to PDF and has the 'documentclass: scrartcl' renders an error regarding the abstract. Unfortunately, the logs are totally uninformative. Compilation works when the 'documentclass' is set to 'article'. 

Weirdly enough, it worked before without any problems, but suddenly all my documents that use scrartcl are failing the compilation.

I would appreciate any help. 

Minimal example:
````qmd

---
title: "Untitled"
format: pdf
---


````

**The terminal output:**

```

PS C:\Users\nikod\Documents> quarto preview c:/Users/nikod/Documents/Untitled-1.qmd --no-browser --no-watch-inputs
pandoc 
  to: latex
  output-file: Untitled-1.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  title: Untitled
  

Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
Argument of \strip@prefix has an extra }.
<inserted text>
                \par
l.3178 ...nglish,usenglish}\abstractname{Abstract}

```

Additional error info after running `$ENV:QUARTO_PRINT_STACK="true"`:
```
ERROR: Error
    at renderFiles (file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:77074:29)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async render (file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:81807:21)
    at async Command.fn (file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:81954:32)
    at async Command.execute (file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:8104:13)
    at async quarto (file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:114999:5)
    at async file:///C:/Users/nikod/AppData/Local/Programs/Quarto/bin/quarto.js:115017:9

```

The result of `quarto check`
```
PS C:\Users\nikod\Documents> quarto check
Quarto 1.4.550
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.550
      Path: C:\Users\nikod\AppData\Local\Programs\Quarto\bin
      CodePage: 1250

[>] Checking tools....................OK
      TinyTeX: v2023.11
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\nikod\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.12.0
      Path: C:/Users/nikod/AppData/Local/Programs/Python/Python312/python.exe
      Jupyter: 5.3.2
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.3.1
      Path: C:/PROGRA~1/R/R-43~1.1
      LibPaths:
        - C:/Users/nikod/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.1/library
      knitr: 1.44
      rmarkdown: 2.25

[>] Checking Knitr engine render......OK

```

**Setup info:**

quarto 1.4.550
Windows 11

**Comment by mcanouil** [comment]
Could you properly format your code to be legible and usable?
~~Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository?~~
Could you share `quarto check` output?
Thank you!

PS: I cannot reproduce on Linux and macOS. Also, if this was really failing, Quarto tests wound have catch the error if the only thing needed was `format: pdf`, thus the issue lies in your setup/environment.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by Niklewa** [reply]
Thanks for the answer. I have edited the description according to your message. `quarto check` seems fine.

**Reply by mcanouil** [reply]
Did you try in other directory?
Because, I cannot reproduce at all for now.

**Reply by Niklewa** [reply]
Yup, it didn't work. It was caused by the setup, problem solved. Thank you for your patience!

**Comment by Niklewa** [comment, marked as answer]
Problem solved, compilation works.

The root of the problem remains unknown, but reinstalling the LaTeX environment resolved the issue. Unfortunately, the error messages and logs were too uninformative to diagnose precisely what was wrong with the setup.

The solution:

First, I removed TinyTeX, ensuring that all files were deleted by checking the package localization:

```
> tinytex::uninstall_tinytex()
```

Then, I reinstall it:

```
> tinytex::install_tinytex()
```

Next, I installed koma-script as it contains `scrartcl` document class (I think):

```
tlmgr install koma-script
```

Then, I updated tlmgr:

```
tlmgr update --self
```

Finally, I proceeded to install all the packages necessary to run the compilation by attempting to compile the document and installing any missing packages indicated by errors.

**Reply by mcanouil** [reply]
This works but I believe there is an easiest approach described in https://quarto.org/docs/output-formats/pdf-engine.html#installing-tex

Quarto has utilities to install, remove, and update TinyTex which does not require R or manual use of `tlmgr`.

Anyhow, glad you fixed your environment.

**Reply by Niklewa** [reply]
I wasn't aware of that, it looks promising. Yes, that's probably an easier and more elegant solution.
